Extensions/ArtifactEngine/Engine/artifactEngine.ts

```diff
@@ -114,8 +115,33 @@ export class ArtifactEngine {
         }
     }
 
+    shouldDownload(path: string) {
+        var shouldDownload = false;
+        for (let pattern of this.patternList) {
+            if (pattern.charAt(0) === '-' && (minimatch(path, pattern.substr(1), { dot: true, nocase: true }))) {
+                return false;
```

Comment:
looks like exclude will override include in any case. this isn't true.

let's say, we have 2 specs:
- exclude c:\* 
- include c:\foo

i suppose we need to download c:\foo\bar, right?

can we do the following:

check if item matches any include
  - false if no include matches
if any include matches, check if item matches any exclude
 - true if no exclude matches
if any exclude matched check if include is a child of exclude 
 - true if include is child of exclude, else false